File: backend/app/api/schemas/notifications.py
from pydantic import BaseModel
from typing import List, Optional, Literal

class NotificationUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    message_type: Optional[str] = None
    target_type: Optional[str] = None

    event_ids: Optional[List[int]] = None
    session_ids: Optional[List[int]] = None
    target_user_ids: Optional[List[int]] = None


class NotificationCreate(BaseModel):
    message_type: Literal["announcement", "alert", "reminder", "warning"]
    target_type: Literal["all", "event", "session", "user"]

    title: str
    content: str

    event_ids: Optional[List[int]] = None
    session_ids: Optional[List[int]] = None
    target_user_ids: Optional[List[int]] = None


class NotificationPublish(BaseModel):
    notification_ids: List[int]

File: frontend/src/services/notifications.ts
// frontend/src/services/notifications.ts
import { apiFetch } from "@/lib/api";

export interface Notification {
  id: number;
  title: string;
  content: string;
  target_type: string;
  message_type: string;
  is_active: boolean;
  published_at: string | null;
  created_at: string;
}

export interface CreateNotificationPayload {
  title: string;
  content: string;
  message_type: "announcement" | "alert" | "reminder" | "warning";
  target_type: "all" | "event" | "session" | "user";
  event_ids?: number[];
  session_ids?: number[];
  target_user_ids?: number[];
}

export const getNotifications = () => {
  return apiFetch("/api/v1/notifications");
};

export const createNotificationDraft = (payload: CreateNotificationPayload) => {
  return apiFetch("/api/v1/notifications", {
    method: "POST",
    body: JSON.stringify(payload),
  });
};

export const publishNotifications = (notification_ids: number[]) => {
  return apiFetch("/api/v1/notifications/publish", {
    method: "POST",
    body: JSON.stringify({ notification_ids }),
  });
};

export const updateNotificationDraft = (
  notificationId: number,
  payload: Partial<Notification>
) => {
  return apiFetch(`/api/v1/notifications/${notificationId}`, {
    method: "PUT",
    body: JSON.stringify(payload),
  });
};
